### PR TITLE
Add torch Imageclassification automodel

### PR DESCRIPTION
**Pull request by AmosLewis**
Add new torch modes:
google/vit-base-patch16-224,True,hf_img_cls,False
microsoft/resnet-50,True,hf_img_cls,False
facebook/deit-small-distilled-patch16-224,True,hf_img_cls,False
microsoft/beit-base-patch16-224-pt22k-ft22k,True,hf_img_cls,False
nvidia/mit-b0,True,hf_img_cls,False

### CODE REVIEW

File: tank/model_utils.py [hunk 1]
```diff
@@ -24,6 +24,65 @@ def get_torch_model(modelname):
...
+        hf_model_path = f"{transformers_path}/models/{hf_model_name}"
+        self.model = AutoModelForImageClassification.from_pretrained(
+            hf_model_name,  # The pretrained model.
+            output_attentions=False,  # Whether the model returns attentions weights.
+            return_dict=False,  # https://github.com/huggingface/transformers/issues/9095
+            torchscript=True,
+        )
+
+    def forward(self, inputs):
+        return self.model.forward(inputs)
```
Comment (pashu123) [inline tank/model_utils.py:75]: self.model.forward(inputs)[0]

Comment (AmosLewis) [reply tank/model_utils.py:75]: I tried. But then I got some error else.

Comment (AmosLewis) [reply tank/model_utils.py:75]: Traceback (most recent call last):
  File "/home/chi/src/ubuntu20/shark/SHARK/generate_sharktank.py", line 221, in <module>
    save_torch_model(args.torch_model_csv)
  File "/home/chi/src/ubuntu20/shark/SHARK/generate_sharktank.py", line 91, in save_torch_model
    mlir_importer.import_debug(
  File "/home/chi/src/ubuntu20/shark/SHARK/shark/shark_importer.py", line 163, in import_debug
    imported_mlir = self.import_mlir(
  File "/home/chi/src/ubuntu20/shark/SHARK/shark/shark_importer.py", line 109, in import_mlir
    return self._torch_mlir(is_dynamic, tracing_required), func_name
  File "/home/chi/src/ubuntu20/shark/SHARK/shark/shark_importer.py", line 74, in _torch_mlir
    return get_torch_mlir_module(
  File "/home/chi/src/ubuntu20/shark/SHARK/shark/torch_mlir_utils.py", line 65, in get_torch_mlir_module
    module = torch_mlir.compile(
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch_mlir/__init__.py", line 186, in compile
    scripted = torch.jit.trace(model, tuple(example_args_for_trace))
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/jit/_trace.py", line 750, in trace
    return trace_module(
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/jit/_trace.py", line 967, in trace_module
    module._c._create_method_from_trace(
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1185, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1173, in _slow_forward
    result = self.forward(*input, **kwargs)
  File "/home/chi/src/ubuntu20/shark/SHARK/tank/model_utils.py", line 75, in forward
    return self.model.forward(inputs)[0]
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/transformers/models/vit/modeling_vit.py", line 765, in forward
    outputs = self.vit(
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1185, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1173, in _slow_forward
    result = self.forward(*input, **kwargs)
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/transformers/models/vit/modeling_vit.py", line 571, in forward
    embedding_output = self.embeddings(
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1185, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1173, in _slow_forward
    result = self.forward(*input, **kwargs)
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/transformers/models/vit/modeling_vit.py", line 134, in forward
    embeddings = self.patch_embeddings(pixel_values, interpolate_pos_encoding=interpolate_pos_encoding)
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1185, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1173, in _slow_forward
    result = self.forward(*input, **kwargs)
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/transformers/models/vit/modeling_vit.py", line 187, in forward
    raise ValueError(
ValueError: Input image size (7*7) doesn't match model (224*224).

Comment (AmosLewis) [reply tank/model_utils.py]: > self.model.forward(inputs)[0]

test_input.shape:  torch.Size([1, 3, 224, 224])
Chi pixel_values.shape:  torch.Size([1, 3, 224, 224])
Chi image_size:  224
actual_out.shape：  torch.Size([1, 1000])


Chi pixel_values.shape:  torch.Size([1, 3, 224, 224])
Chi image_size:  224
/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/transformers/models/vit/modeling_vit.py:188: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if height != self.image_size[0] or width != self.image_size[1]:
Chi pixel_values.shape:  torch.Size([1, 3, 224, 224])
Chi image_size:  224
/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/jit/_trace.py:443: UserWarning: CUDA initialization: Unexpected error from cudaGetDeviceCount(). Did you run some cuda functions before calling NumCudaDevices() that might have already set an error? Error 804: forward compatibility was attempted on non supported HW (Triggered internally at ../c10/cuda/CUDAFunctions.cpp:109.)
  outs = wrap_retval(mod(*_clone_inputs(inputs)))
Chi pixel_values.shape:  torch.Size([1, 3, 224, 224])
Chi image_size:  224
Chi pixel_values.shape:  torch.Size([1, 3, 224, 224])
Chi image_size:  224
Chi pixel_values.shape:  torch.Size([7, 7, 7, 7])
Chi image_size:  224
Traceback (most recent call last):
  File "/home/chi/src/ubuntu20/shark/SHARK/generate_sharktank.py", line 221, in <module>
    save_torch_model(args.torch_model_csv)
  File "/home/chi/src/ubuntu20/shark/SHARK/generate_sharktank.py", line 91, in save_torch_model
    mlir_importer.import_debug(
  File "/home/chi/src/ubuntu20/shark/SHARK/shark/shark_importer.py", line 163, in import_debug
    imported_mlir = self.import_mlir(
  File "/home/chi/src/ubuntu20/shark/SHARK/shark/shark_importer.py", line 109, in import_mlir
    return self._torch_mlir(is_dynamic, tracing_required), func_name
  File "/home/chi/src/ubuntu20/shark/SHARK/shark/shark_importer.py", line 74, in _torch_mlir
    return get_torch_mlir_module(
  File "/home/chi/src/ubuntu20/shark/SHARK/shark/torch_mlir_utils.py", line 65, in get_torch_mlir_module
    module = torch_mlir.compile(
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch_mlir/__init__.py", line 186, in compile
    scripted = torch.jit.trace(model, tuple(example_args_for_trace))
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/jit/_trace.py", line 750, in trace
    return trace_module(
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/jit/_trace.py", line 967, in trace_module
    module._c._create_method_from_trace(
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1185, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1173, in _slow_forward
    result = self.forward(*input, **kwargs)
  File "/home/chi/src/ubuntu20/shark/SHARK/tank/model_utils.py", line 75, in forward
    return self.model.forward(inputs)[0]
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/transformers/models/vit/modeling_vit.py", line 767, in forward
    outputs = self.vit(
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1185, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1173, in _slow_forward
    result = self.forward(*input, **kwargs)
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/transformers/models/vit/modeling_vit.py", line 573, in forward
    embedding_output = self.embeddings(
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1185, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1173, in _slow_forward
    result = self.forward(*input, **kwargs)
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/transformers/models/vit/modeling_vit.py", line 134, in forward
    embeddings = self.patch_embeddings(pixel_values, interpolate_pos_encoding=interpolate_pos_encoding)
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1185, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1173, in _slow_forward
    result = self.forward(*input, **kwargs)
  File "/home/chi/src/ubuntu20/shark/SHARK/shark.venv/lib/python3.10/site-packages/transformers/models/vit/modeling_vit.py", line 189, in forward
    raise ValueError(
ValueError: Input image size (7*7) doesn't match model (224*224).

Comment (AmosLewis) [reply tank/model_utils.py]: Fixed by adding a dynamic flag in csv.

**Review by monorimet — APPROVED** [review]
LGTM

**Pull request merged**